usr.sbin/daemon/daemon.c

```diff
@@ -341,46 +342,19 @@ main(int argc, char *argv[])
 	pidfile_write(state.parent_pidfh);
 
 	if (supervision_enabled) {
-		struct sigaction act_term = { 0 };
-		struct sigaction act_chld = { 0 };
-		struct sigaction act_hup = { 0 };
-
-		/* Avoid PID racing with SIGCHLD and SIGTERM. */
-		act_term.sa_handler = handle_term;
-		sigemptyset(&act_term.sa_mask);
-		sigaddset(&act_term.sa_mask, SIGCHLD);
-
-		act_chld.sa_handler = handle_chld;
-		sigemptyset(&act_chld.sa_mask);
-		sigaddset(&act_chld.sa_mask, SIGTERM);
-
-		act_hup.sa_handler = handle_hup;
-		sigemptyset(&act_hup.sa_mask);
 
 		/* Block SIGTERM to avoid racing until we have forked. */
```

Comment:
why not move this one too?

Reply:
setting signal handlers happens once at the very beginning of the supervisor's startup.

the usage of `sigprocmask` is different, daemon does `SIG_UNBLOCK` as soon as fork succeeds later in this function. So this does not belong at `setup_signals()` I think keeping BLOCK/UNBLOCK pair in the same function is better for clarity.

See:  line 419
https://github.com/freebsd/freebsd-src/pull/684/files/f3f7fc66d65e178f8f53e0987d3bedca58604d59..2cbe6eb7ffab54ee544e213ede53a328ae3909e0#diff-e92ba17e18b6d814df32ae1bcf967fa671e44757395813d70a5305b74b556ff8R419